src/main/java/edu/nicholaidudakiwwarrick/advancedjava/services/ServiceFactory.java

```diff
@@ -0,0 +1,41 @@
+package edu.nicholaidudakiwwarrick.advancedjava.services;
+
+import jdk.nashorn.internal.ir.annotations.Immutable;
+
+/**
+ * Factory pattern class to instantiate service objects
+ *
+ * @author Nicholai Dudakiw-Warrick
+ */
+@Immutable
+public final class ServiceFactory {
+
+    /**
+     * Private constructor to prevent instantiation of static factory instances
+     */
+    private ServiceFactory() {
+    }
+
+    /**
+     * @param type {@code ServiceType} enum that specifies the type of {@code StockService} to return
+     * @return the {@code BasicStockService} or {@code DatabaseStockService} instance as a reference to
+     * the {@code StockService} interface
+     */
```

Comment:
be sure to document the exception 
@throws  StockServiceException 
be sure to include why it might happen and what the calling code might be able to do to resolve the problem.